### PR TITLE
Added: Add score feedback display options to MatchingPairsGame

### LINKED ISSUE #716: Small Visual Feedback Option
Related to #605 and #609 and #603.

All audio versions of the Congo Branch (#603) should have version where feedback options are shown in small text, in bottom right corner of screen. The purpose of this is so that the player (member of BaYaka group) can play the game without having feedback that is meaningful to them, but the experimenter (@Lia-JX-Li ) can visually confirm that the game is working as expected. 

The final solution will look like the normal version of the game, with the feedback location and size changed. See attached (but smaller text).

Screenshot 2024-01-17 at 10 21 35

### PR DESCRIPTION
**Pull request by drikusroor**
This pull request adds the functionality to display score feedback in the MatchingPairsGame component. It includes the following changes:

- Added a new prop called scoreFeedbackDisplay to the MatchingPairs component.

- Refactored the MatchingPairs component to support the scoreFeedbackDisplay prop.

- Added a new option called SCORE_FEEDBACK_DISPLAY to the MatchingPairs component.

- Updated the WithSmallBottomRightScoreFeedback story to demonstrate the new score feedback display option.

Resolves #716

### CODE REVIEW

File: backend/experiment/rules/matching_pairs.py
Comment (BeritJanssen) [inline backend/experiment/rules/matching_pairs.py:114]: Be sure to remove this modification before merging.

**Review by BeritJanssen — APPROVED** [review]
Looks good to me! Let's merge this after #672 (I'll merge this today).

**Comment by BeritJanssen** [comment]
FYI, on develop the "score_feedback_display" is now called "display_score", and is a kwarg on `experiment.rules.actions.playback.MatchingPairs` initialization. Feel free to move it up to `Playback` instead, but since we usually don't include scores in players, I thought to move it only to the MatchingPairs player types. Also, feel free to rename. I tried to emulate your options. Perhaps the backend should also get a definition of the different options that are allowed for the display options? We might add them as "score display types" to the top of `playback.py` and adjust typing so it only allows one of the three strings?

**Comment by drikusroor** [comment]
> FYI, on develop the "score_feedback_display" is now called "display_score", and is a kwarg on `experiment.actions.playback.MatchingPairs` initialization. Feel free to move it up to `Playback` instead, but since we usually don't include scores in players, I thought to move it only to the MatchingPairs player types. Also, feel free to rename. I tried to emulate your options. Perhaps the backend should also get a definition of the different options that are allowed for the display options? We might add them as "score display types" to the top of `playback.py` and adjust typing so it only allows one of the three strings?



> FYI, on develop the "score_feedback_display" is now called "display_score", and is a kwarg on `experiment.actions.playback.MatchingPairs` initialization. Feel free to move it up to `Playback` instead, but since we usually don't include scores in players, I thought to move it only to the MatchingPairs player types. Also, feel free to rename. I tried to emulate your options. Perhaps the backend should also get a definition of the different options that are allowed for the display options? We might add them as "score display types" to the top of `playback.py` and adjust typing so it only allows one of the three strings?

Huh? Is this new or did I just overlook the existing prop completely?

**Comment by BeritJanssen** [comment]
This is new, after the merge of Evert's pull request, which kinda but not quite addressed a similar request.

**Review by BeritJanssen — APPROVED** [review]

**Pull request merged**
